src/it/resources/com/google/checkstyle/test/chapter5naming/rule528typevariablenames/InputInterfaceTypeParameterName.java

```diff
@@ -41,41 +44,54 @@ <T> void getMoreFoo() {}
   }
 }
 
+// violation below 'Top-level class Boo has to reside in its own source file.'
 interface Boo<Input> { // violation 'Interface type name 'Input' must match pattern'
   Input boo();
 }
```

Comment:
moving extra interfaces/classes inside main class would make the file not compile-able so had to add these violations